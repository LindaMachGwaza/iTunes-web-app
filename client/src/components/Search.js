import React from 'react'
// Import react-bootstrap components
import { Button, Form, FormControl, FormGroup, FormSelect } from 'react-bootstrap'
// Import CSS
import '../App.css'

const Search =({
    handleSubmit,
    term,
    handleTermChange,
    handleMediaChange
}) => {
    //Media type array
    const media = [
        {type: 'All media', value: 'all'},
        {type: 'Movie', value: 'movie'},
        {type: 'Podcast', value: 'podcast'},
        {type: 'Music', value: 'music'},
        {type: 'Music video', value: 'musicVideo'},
        {type: 'Audio book', value: 'audiobook'},
        {type: 'Short film', value: 'shortFilm'},
        {type: 'TV show', value: 'tvShow'},
        {type: 'Software', value: 'software'},
        {type: 'Ebook', value: 'ebook'}
     ];

     return (
        <div className='search-container'>
            <Form onSubmit={handleSubmit}> {/*Input field for searching; user enters a term they want to search for */}
                <FormGroup className='form-group search-box'>
                    <FormControl type='text' className='search-bar'placeholder='Search...'
                    name='term'value={term} onChange={handleTermChange}
                    />
                </FormGroup>
                <FormGroup className='form-group filter'>
                <FormSelect onChange={handleMediaChange}> {/*Display media options dropdown menu */}
                    {media && media.map((media)=> (
                        <option key ={media.value} value ={media.value}>
                                {media.type}
                        </option>
                    ))}
                </FormSelect>
                </FormGroup>
                <FormGroup className='submit-btn'>{/*Search items on button click */}
                    <Button variant='secondary' type='submit'>
                        Search
                    </Button>
                </FormGroup>
            </Form>

        </div>
     )
}

export default Search;

/*Sources used in executing the task include Hyperiondev notes and vedios, previous tasks, You Tube, Google.com, 
StackOverflow, Github, Geeksforgeeks*/