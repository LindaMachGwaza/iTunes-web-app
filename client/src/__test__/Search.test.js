import renderer from 'react-test-renderer';
import Search from '../components/Search';
//create a snapshot of the Search component
test('to match Search snapshot', () =>{
    const tree = renderer.create(<Search/>).toJSON();

    expect (tree).toMatchSnapshot();
})