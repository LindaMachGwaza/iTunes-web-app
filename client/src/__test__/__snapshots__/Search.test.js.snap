// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match Search snapshot 1`] = `
<div
  className="search-container"
>
  <form
    className=""
  >
     
    <div
      className="form-group search-box"
    >
      <input
        className="search-bar form-control"
        name="term"
        placeholder="Search..."
        type="text"
      />
    </div>
    <div
      className="form-group filter"
    >
      <select
        className="form-select"
      >
         
        <option
          value="all"
        >
          All media
        </option>
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="podcast"
        >
          Podcast
        </option>
        <option
          value="music"
        >
          Music
        </option>
        <option
          value="musicVideo"
        >
          Music video
        </option>
        <option
          value="audiobook"
        >
          Audio book
        </option>
        <option
          value="shortFilm"
        >
          Short film
        </option>
        <option
          value="tvShow"
        >
          TV show
        </option>
        <option
          value="software"
        >
          Software
        </option>
        <option
          value="ebook"
        >
          Ebook
        </option>
      </select>
    </div>
    <div
      className="submit-btn"
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</div>
`;
